torch/distributed/algorithms/ddp_comm_hooks/optimizer_overlap_hooks.py

```diff
@@ -21,9 +22,20 @@ def __init__(
             **functional_optim_kwargs,
             _allow_empty_param_list=True,
         )
-        if not hasattr(self.functional_optimizer, "step_param"):
+        self._check_valid_functional_optim()
+
+    @classmethod
+    def from_functional_optim(cls, functional_optim):
+        opt_hook_state_inst = cls.__new__(cls)  # Does not call __init__
+        opt_hook_state_inst.functional_optimizer = functional_optim
+        opt_hook_state_inst._check_valid_functional_optim()
+        return opt_hook_state_inst
+
+    def _check_valid_functional_optim(self):
+        if not hasattr(self.functional_optimizer, _FUNCTIONAL_OPTIM_STEP_METHOD_NAME):
```

Comment:
Should we add a brief docstr to `from_functional_optim` that describes this requirement? I see that we already mention this in `__init__`'s description.

Reply:
Added this in the latest update, although I later realized that I should just enhance the doc in the top diff, as this is just moved to a regular `__init__` at the top diff.